src/mbgl/map/map.cpp

```diff
@@ -727,44 +625,30 @@ void Map::updateAnnotation(AnnotationID id, const Annotation& annotation) {
 
 void Map::removeAnnotation(AnnotationID annotation) {
     impl->annotationManager.removeAnnotation(annotation);
-    impl->onUpdate(Update::AnnotationStyle | Update::AnnotationData);
+    impl->onUpdate(Update::AnnotationData);
 }
 
 #pragma mark - Feature query api
 
 std::vector<Feature> Map::queryRenderedFeatures(const ScreenCoordinate& point, const RenderedQueryOptions& options) {
```

Comment:
Can we do with `Map::query*Features` what we did for `Map::onLowMemory`, removing them from the `Map` interface and replacing calls to them with calls to methods on a concrete `RendererFrontend` classes?